M365/MDO/ResendFailedMail.ps1

```diff
@@ -231,7 +231,7 @@ if ($failedMessages.count -ge 1000) {
 }
 
 if (-not $IncludeDuplicates) {
-    [array]$failedMessages = $failedMessages | Sort-Object MessageId -Unique
+    [array]$failedMessages = $failedMessages | Group-Object -Property MessageId | ForEach-Object { $_.Group | Sort-Object -Property Received | Select-Object -First 1 }
```

Comment:
How does this address empty subjects?

Reply:
It looks an issue with Get-MessageTrace:
for all original messages it is able to get subject, but for some resend messages is not able.
With this we will get the oldest one, the original with that messageId

Reply:
Would it be better to include a where statement when the subject isn't null?

Reply:
I though that but if the original message is null, it will not work

Reply:
We talking about original message subject or original message MessageId